src/net_processing.cpp

```diff
@@ -4353,7 +4421,7 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         } // cs_main
 
         if (fProcessBLOCKTXN) {
-            return ProcessMessage(pfrom, NetMsgType::BLOCKTXN, blockTxnMsg, time_received, interruptMsgProc);
+            return ProcessCompactBlockTxns(pfrom, txn, *peer, msgMaker);
```

Comment:
Can't this call be moved up where `fProcessBLOCKTXN=true`, and this variable removed altogether, in a pure refactoring way?

Reply:
Can we? would be ok to call `ProcessCompactBlockTxns` into `cs_main` block?

Reply:
I think so. `fProcessBLOCKTXN` exists to call `ProcessMessage` out of `cs_main` block, but I think we could call `ProcessCompactBlockTxns` there.

Reply:
ahhhh you're right, it's the `cs_main` thing.
An easy refactoring (splitting `ProcessBlock` out of `ProcessCompactBlockTxns`) would allow avoid one unlock/lock operation, but I'm not sure it's worth it.